### ISSUE TITLE
Parameter values with "-" character raise ValueError

### ISSUE THREAD

**Issue by Chryron** [BuildACell/BioCRNPyler#171]
If any parameter values contain the "-" character (e.g "1e-4"), the error below is triggered 
```python
if re.search('[a-d-f-z]', new_parameter_value, re.I) \
        or re.search('(^[1-9]+/[1-9]+)|(^[1-9]+e-?[0-9]+)|(^.?[0-9])', new_parameter_value, re.I) is None:
    raise ValueError(f'No valid parameter value! Accepted formats: 1.00 or 1e4 or 2/5, we got {new_parameter_value}')
```
since the `[a-d-f-z]` regex expression catches "-". The fix is just to change `[a-d-f-z]` to `[a-df-z]`

**Comment by WilliamIX** [comment]
This is correct. SBML does not allow variables (including parameters) to use "-". Please use different parameter names.

**Comment by Chryron** [comment]
Ah so instead of having a parameter be 1e-6, its only acceptable to use 0.000001?

**Comment by ayush9pandey** [comment]
I think Nazmus meant that for parameter values (and the negative powers)

**Comment by WilliamIX** [comment]
@Chryron are you passing a parameter directly to a reaction with a string for the parameter value or is this through a parameter file or parameter dictionary?

**Comment by Chryron** [comment]
Through a parameter file but I can try the other ways. I believe the error is raised whenever its a string.

**Comment by zoltuz** [comment]
Hi @Chryron, thanks for reporting this. This issue has been solved in PR #178 (thanks @WilliamIX ) and the dev branch now contains the bugfix.

If you want to give a try, you can install biocrnpyler using the code from the dev branch:
https://github.com/BuildACell/BioCRNPyler/wiki/How-to-install-BioCRNPyler#installing-the-dev-branch
